deephub/trainer/__main__.py

```diff
@@ -114,7 +115,12 @@ def train(variant_name, variants_dir, gpu_device_ids, extra_expr_params, extra_s
         variant_definition.set('train.warm_start_check_point', warm_start_checkpoint)
         variant_definition.set('train.warm_start_variables_regex', warm_start_vars)
 
-    trainer = Trainer(requested_gpu_devices=gpu_device_ids)
+    # avoid passing "-1" with the regular arguments.
+    if len(gpu_device_ids) == 1 and gpu_device_ids[0] == -1:
+        gpu_device_ids = tuple(find_idle_gpu_devices())
```

Comment:
is tuple conversion really needed?

Reply:
yeap, since the multiple -g arguments are passed as tuple instead of a list.